packages/mosaic-tanstack-table-core/src/facet-menu.ts

```diff
@@ -0,0 +1,404 @@
+// packages/mosaic-tanstack-table-core/src/facet-menu.ts
+
+import {
+  MosaicClient,
+  coordinator as defaultCoordinator,
+} from '@uwdata/mosaic-core';
+import * as mSql from '@uwdata/mosaic-sql';
+import { Store } from '@tanstack/store';
+import { createStructAccess } from './utils';
+import { logger } from './logger';
+import type { Coordinator, Selection } from '@uwdata/mosaic-core';
+import type { FilterExpr, SelectQuery } from '@uwdata/mosaic-sql';
+
+export interface MosaicFacetMenuOptions {
+  table: string;
+  column: string;
+  selection: Selection;
+  filterBy?: Selection;
+  additionalContext?: Selection;
+  coordinator?: Coordinator;
+  sortMode?: 'alpha' | 'count';
+  limit?: number;
+  debugName?: string;
+  isArrayColumn?: boolean;
+}
+
+export interface MosaicFacetMenuState {
+  options: Array<any>;
+  loading: boolean;
+  searchTerm: string;
+}
+
+let instanceCounter = 0;
+
+export class MosaicFacetMenu extends MosaicClient {
```

Comment:
This needs a comment to explain why this client is needed and what it does in relation to the data table client.